### PR TITLE
Cleaned up bintray provider pull request

### PR DESCRIPTION
**Pull request by rtyler**
This is just a bit of rebasing and tidying up of #255, which can be closed now.

It contains only @eyalbe4's original changes, I just rebased the commits together a bit and updated some commit messages.

### CODE REVIEW

File: lib/dpl/provider/bintray.rb [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+      @descriptor
+
+      def check_auth
+      end
+
+      def needs_key?
+        false
+      end
+
+      def initFromArgs
```
Comment (BanzaiMan) [inline lib/dpl/provider/bintray.rb]: Please use snake case, which is more in line with Ruby conventions.

File: lib/dpl/provider/bintray.rb [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+
+      def readDescriptor
+        log "Reading descriptor file: #{@file}"
+        file = File.open(@file)
+        content = ""
+        file.each {|line|
+          content << line
+        }
+
+        @descriptor = JSON.parse(content)
```
Comment (BanzaiMan) [inline lib/dpl/provider/bintray.rb]: Seems a rather roundabout way to read the content of `@file`. You can just do `JSON.parse(File.read(@file))`.

File: lib/dpl/provider/bintray.rb [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+        subject = package["subject"]
+        repo = package["repo"]
+        versionName = version["name"]
+
+        path = "/content/#{subject}/#{repo}/#{packageName}/#{versionName}/#{artifact.getUploadPath}"
+        res = putBinaryFileRequest(artifact.getLocalPath, path, artifact.getMatrixParams)
+        logBintrayResponse(res)
+      end
+
+      def packageExists
```
Comment (BanzaiMan) [inline lib/dpl/provider/bintray.rb]: I suggest `package_exists?` as a method name.

File: lib/dpl/provider/bintray.rb [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+          return false
+        end
+        if code == 201 || code == 200
+          return true
+        end
+        abort("Unexpected HTTP response code #{code} returned from Bintray while checking if package '#{name}' exists. " +
+          "Response message: #{res.message}")
+      end
+
+      def versionExists
```
Comment (BanzaiMan) [inline lib/dpl/provider/bintray.rb]: `version_exists?`

File: lib/dpl/provider/bintray.rb [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+        files = getFilesToUpload
+
+        files.each do |key, artifact|
+          uploadFile(artifact)
+        end
+      end
+
+      def publishVersion
+        publish = @descriptor["publish"]
+        if publish
```
Comment (BanzaiMan) [inline lib/dpl/provider/bintray.rb]: Is this right? Can `publish` have values such as `"NO"` or `"false"`?

**Comment by BanzaiMan** [comment]
Thanks for the PR. Could you add specs, pleas?

**Comment by rtyler** [comment]
@eyalbe4 will you be able to add the RSpec for this?

**Comment by eyalbe4** [comment]
@rtyler , Sure. I'm working on adding the specs and implementing the other comments.
Question: Since you submitted this PR, how do you suggest we add my new content?

**Comment by BanzaiMan** [comment]
@eyalbe4: @rtyler can make you a collaborator to his fork, so that you can push to this PR.

**Comment by rtyler** [comment]
@eyalbe4 you should be able to add changes to the bintray-provider branch,in my repo. That said, if you'd like me to help with reviewing the specs before we take up @BanzaiMan's time agian, you can also submit a pull request to that branch as well.

**Comment by eyalbe4** [comment]
Thank you so much @rtyler .
I'll do that.

**Comment by eyalbe4** [comment]
@rtyler , I committed the changes into your repo.
It includes the specs addition and implementation of the comments provided by @BanzaiMan .
As for the questions raised by @BanzaiMan :
'Is this right? Can publish have values such as "NO" or "false"?'
Yes - the "publish" contains a Boolean value read from the json descriptor file:
@rtyler and @bobzoller  - thanks again for your help promoting this!

**Comment by eyalbe4** [comment]
Hi @rtyler ,
Did you get a chance to look at the specs I added?

File: spec/provider/bintray_spec.rb [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+  end
+
+  def init_provider(bintray)
+    bintray.init_from_args
+    bintray.set_descriptor(get_descriptor_content)
+    bintray.set_test_mode
+  end
+
+  def get_descriptor_content
+    return '{'\
```
Comment (rtyler) [inline spec/provider/bintray_spec.rb]: For future reference, Ruby supports multiline strings with the `"""` denotation, e.g.:

```
2.1.5 :001 > foo = """
2.1.5 :002"> this
2.1.5 :003"> is  
2.1.5 :004"> sparta!
2.1.5 :005"> """
 => "\nthis\nis\nsparta!\n" 
2.1.5 :006 > puts foo

this
is
sparta!
 => nil 
2.1.5 :007 > 
``

File: spec/provider/bintray_spec.rb [hunk 7]
```diff
@@ -0,0 +1,261 @@
+require 'spec_helper'
+require 'dpl/provider/bintray'
+
+describe DPL::Provider::Bintray do
+  @@package_name
```
Comment (rtyler) [inline spec/provider/bintray_spec.rb]: What's the point of `package_name`?

**Comment by rtyler** [comment]
For the most part *I* think this is fine, but this isn't my repo, it's @BanzaiMan's :)

**Comment by eyalbe4** [comment]
Thanks a lot for the review @rtyler !
I implemented your 2 comments and pushed them to your repo.
Could you add this to this pull request?

**Comment by eyalbe4** [comment]
I created a new pull request: https://github.com/travis-ci/dpl/pull/259
containing all comments provider by @rtyler and @BanzaiMan
so @rtyler, I think you can close this one.

**Pull request closed without merging**